File: python-net/merge/Merger_pdf_bmp.py
from Aspose.Imaging.FileFormats.Bmp import BmpImage
from Aspose.Imaging.ImageOptions import BmpOptions
from Aspose.Imaging.Sources import FileCreateSource
from Aspose.Imaging import Rectangle
from Aspose.Imaging import Source
from Aspose.Imaging import RasterImage
from Aspose.Imaging import Size
from Aspose.Imaging import Image
from Aspose.Pdf.Devices import BmpDevice
from Aspose.Pdf.Devices import Resolution
from Aspose.Pdf import Document
import clr

aspose_pdf = clr.AddReference("../../lib/Aspose.PDF.dll")
aspose_imaging = clr.AddReference("../../lib/Aspose.Imaging.dll")

class pdf_to_bmp(object):
    def __init__(self, licence_path):
        self.dataDir = "../../TestData"
        if licence_path:
            self.licence_path = licence_path
            self.aspose_license = License()
            self.aspose_license.SetLicense(self.licence_path)

    def exec(self):
        pathSource = "../../TestData/test.pdf"

        # read pdf file to Aspose Document
        doc = Document(pathSource)

        # make list of path to temporary images
        images = []

        # pages in pdf counted from 1 to n
        for pageCount in range(1, doc.Pages.Count):
            # setup default resolution to pdf documents 72dpi
            resolution = Resolution(72)

            # create image device to save document as image with page dimensions and resolution
            imageDevice = BmpDevice(
                doc.Pages[pageCount].PageInfo.Width, doc.Pages[pageCount].PageInfo.Height, resolution)

            outPath = "test_"+pageCount+".bmp"

            # process document page to image
            imageDevice.Process(doc.Pages[pageCount], outPath)
            images.Add(outPath)

        # make list pf parsed image sizes
        imageSizes = []
        for path in images:
            # load image from file, it supports a lot of formats
            image = Image.Load(path)
            imageSizes.Add(image.Size)

        newWidth = 0
        newHeight = 0
        for s in imageSizes:
            newWidth += s.Width
            if newHeight < s.Height:
                newHeight = s.Height
            else:
                newHeight = newHeight

        # use file system as source for save image
        # preserve image on the disk
        fileSource = FileCreateSource("./test.bmp", False)

        options = BmpOptions
        options.Source = fileSource

        # create empty image with calculated witdh and hight
        newImage = Image.Create(options, newWidth, newHeight)

        stitchedWidth = 0
        for imagePath in images:
            # load image from file, it supports a lot of formats
            image = Image.Load(imagePath)
            # create bounds to insert small image into large
            bounds = Rectangle(stitchedWidth, 0, image.Width, image.Height)

            # combining images into new one
            # bounds = where to insert image
            # LoadArgb32Pixels = convert image chunk to 32bit Argb
            newImage.SaveArgb32Pixels(
                bounds, image.LoadArgb32Pixels(image.Bounds))
            stitchedWidth += image.Width

        # save created image to disk
        newImage.Save()

File: net/merge/Merger_pdf_bmp.cs
namespace aspose_snippets.net
{
    public static partial class Merger
    {
        public static void pdf_to_bmp()
        {
            var pathSource = @"..\..\TestData\test.pdf";

            //read pdf file to Aspose Document
            var doc = new Aspose.Pdf.Document(pathSource);

            //make list of path to temporary images
            var images = new List<string>();

            //pages in pdf counted from 1 to n
            for (int pageCount = 1; pageCount <= doc.Pages.Count; pageCount++)
            {
                //setup default resolution to pdf documents 72dpi
                var resolution = new Aspose.Pdf.Devices.Resolution(72);

                //create image device to save document as image with page dimensions and resolution
                var imageDevice = new Aspose.Pdf.Devices.BmpDevice(
                        (int)doc.Pages[pageCount].PageInfo.Width, 
                        (int)doc.Pages[pageCount].PageInfo.Height, 
                        resolution);

                var outPath = "test_"+pageCount+".bmp";

                //process document page to image
                imageDevice.Process(doc.Pages[pageCount], outPath);
                images.Add(outPath);
            }

            //make list pf parsed image sizes
            var imageSizes = new List<Aspose.Imaging.Size>();
            foreach(var path in images)
            {
                //load image from file, it suport a lot of formats
                using (Aspose.Imaging.RasterImage image = 
                            (Aspose.Imaging.RasterImage)Aspose.Imaging.Image.Load(path))
                {
                    imageSizes.Add(image.Size);
                }
            }

            int newWidth = imageSizes.Sum(size => size.Width);
            int newHeight = imageSizes.Max(size => size.Height);

            //use file system as source for save image
            Aspose.Imaging.Source fileSource = new Aspose.Imaging.Sources.FileCreateSource(
                "./test.bmp", 
                isTemporal: false); //preserve image on the disk

            var options = new Aspose.Imaging.ImageOptions.BmpOptions() 
                { 
                    Source = fileSource 
                };

            //create empty image with calculated witdh and hight
            using (var newImage = (Aspose.Imaging.FileFormats.Bmp.BmpImage)Aspose.Imaging.Image.Create(options, newWidth, newHeight))
            {
                int stitchedWidth = 0;
                foreach (string imagePath in images)
                {
                    //load image from file, it suport a lot of formats
                    using (var image = (Aspose.Imaging.RasterImage)Aspose.Imaging.Image.Load(imagePath))
                    {
                        //create bounds to insert small image into large
                        Aspose.Imaging.Rectangle bounds = new Aspose.Imaging.Rectangle(
                            stitchedWidth, 
                            0, 
                            image.Width, 
                            image.Height);

                        //combining images into new one
                        newImage.SaveArgb32Pixels(
                            //where to insert image
                            bounds,
                            //convert image chunk to 32bit Argb
                            image.LoadArgb32Pixels(image.Bounds));
                        stitchedWidth += image.Width;
                    }
                }

                //save created image to disk
                newImage.Save();
            }
        }
    }
}
